### PR TITLE
add tests for contactsintegration

### PR DESCRIPTION
**Pull request by ChristophWurst**
ContactsIntegration didn't have any tests: https://scrutinizer-ci.com/g/owncloud/mail/code-structure/master?elementType=class&orderField=test_coverage&order=asc&changesExpanded=0

@itheiss @colmoneill should be easy to review ;-)

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Nice:  Test Coverage 48%▲ +2%

File: tests/service/contactsintegrationtest.php [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		$this->contactsManager->expects($this->never())
+			->method('search');
+
+		$expected = [];
+		$actual = $this->contactsIntegration->getMatchingRecipient("abc");
+
+		$this->assertEquals($expected, $actual);
+	}
+
+	public function testGetMathingRecipient() {
```
Comment (itheiss) [inline tests/service/contactsintegrationtest.php]: Typo? Should be testGetMatchingRecipient

File: tests/service/contactsintegrationtest.php [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		];
+		$actual = $this->contactsIntegration->getMatchingRecipient($term);
+
+		$this->assertEquals($expected, $actual);
+	}
+
+	public function getPhotoDataProvider() {
+		return [
+			[
+				// Math with photo
```
Comment (itheiss) [inline tests/service/contactsintegrationtest.php]: Typo :-) Math => Match

File: tests/service/contactsintegrationtest.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+				'john@doe.com',
+				[
+					'id' => 2,
+					'FN' => 'John Doe',
+					'PHOTO' => 'abcdefg'
+				],
+				'abcdefg',
+			],
+			[
+				// Match withou photo
```
Comment (itheiss) [inline tests/service/contactsintegrationtest.php]: Typo :-)) withou => without

**Comment by itheiss** [comment]
LGTM. Except the little typos.

**Pull request merged**